### PR TITLE
[WIP ]Update to run on CircleCI 2.0

### PR DESCRIPTION
**Pull request by drazisil**

### CODE REVIEW

**Comment by drazisil** [comment]
Failing due to token mismatch, how do you want to handle that?

**Comment by blueyed** [comment]
Would be good to separate into two PRs maybe (fixing tests by itself).
Fixes https://github.com/codecov/codecov-bash/issues/82 then.
But it seems like this repo is not well maintained unfortunately.

**Comment by drazisil** [comment]
I'm trying to improve that (not well maintained). We shall see how much time I can devote to that.

**Comment by blueyed** [comment]
Ah, I see now that we had contact in https://codecov.freshdesk.com/helpdesk/tickets/1990 already.
Thanks!

**Comment by drazisil** [comment]
@blueyed This is mainly to setup the tests on CircleCI, not to fix them.

The changes to the test are because I want to try and get coverage reports on the uploader. Which, to be fair, probably should be a separate PR. I'm very bad at scope creep.

**Comment by drazisil** [comment]
What I'm finding I really don't like here is that the tests are failing like crazy and think they are passing. Not cool :(

**Comment by drazisil** [comment]
This should be good to merge once the repo is added back to CircleCI

I have, in theory, tested this here: https://github.com/drazisil/kcov-bats-circleci-codecov

**Comment by blueyed** [comment]
There is still `circle.yml` in the root.

**Comment by drazisil** [comment]
Waiting to have project added back to CircleCI to actually test before merging

**Comment by drazisil** [comment]
I guess I have to merge _something_ before CircleCI will start building. :crossed_fingers:

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      # per https://github.com/SimonKagstrom/kcov/issues/45#issuecomment-68207360 this should work
+      - run:
+          name: Run Tests
+          command: |
+            kcov --include-path=./codecov coverage/ tests/test
+      - run:
+          name: Upload Codecov
+          command: |
+            bash <(curl -s https://codecov.io/bash) -s coverage -d > upload.txt
+            bash <(curl -s https://codecov.io/bash) -s coverage
```
Comment (blueyed) [inline .circleci/config.yml]: Why is it uploaded twice?

Comment (drazisil) [reply .circleci/config.yml]: It's not. The `-d` flag is debug and output to stdout, so it saves as `upload.txt`. This allows it to be saved as an artifact so I can see what it being sent to the server

Comment (blueyed) [reply .circleci/config.yml]: I see.
It is not included with `store_artifacts` though.

Comment (blueyed) [reply .circleci/config.yml]: Likely `codecov.txt` was meant to be that.
Isn't dumping it to stdout good/better?

Comment (drazisil) [reply .circleci/config.yml]: I'd rather save it as an artifact. But you are right, I renamed tone place and not another

**Pull request merged**